src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs

```diff
@@ -158,6 +235,43 @@ await WaitForUpdatedPullRequestAsync(targetRepoName, targetBranch)
             }
         }
 
+        public async Task CheckBatchedAzDoPullRequest(string source1RepoName, string source2RepoName, string targetRepoName, string targetBranch,
+            List<Microsoft.DotNet.DarcLib.DependencyDetail> expectedDependencies, string repoDirectory, bool complete = false)
+        {
+            string expectedPRTitle = $"[{targetBranch}] Update dependencies from {_parameters.AzureDevOpsAccount}/{_parameters.AzureDevOpsProject}/{source1RepoName} {_parameters.AzureDevOpsAccount}/{_parameters.AzureDevOpsProject}/{source2RepoName}";
+            await CheckAzDoPullRequest(expectedPRTitle, targetRepoName, targetBranch, expectedDependencies, repoDirectory, complete, true, null, null);
+        }
+
+        public async Task CheckNonBatchedAzDoPullRequest(string sourceRepoName, string targetRepoName, string targetBranch,
+            List<Microsoft.DotNet.DarcLib.DependencyDetail> expectedDependencies, string repoDirectory, bool isCompleted = false, bool isUpdated = false,
+            string[] expectedFeeds = null, string[] notExpectedFeeds = null)
+        {
+            string expectedPRTitle = $"[{targetBranch}] Update dependencies from {_parameters.AzureDevOpsAccount}/{_parameters.AzureDevOpsProject}/{sourceRepoName}";
+            await CheckAzDoPullRequest(expectedPRTitle, targetRepoName, targetBranch, expectedDependencies, repoDirectory, false, isUpdated, expectedFeeds, notExpectedFeeds);
+        }
+
+        public async Task CheckAzDoPullRequest(string expectedPRTitle, string targetRepoName, string targetBranch,
+            List<Microsoft.DotNet.DarcLib.DependencyDetail> expectedDependencies, string repoDirectory, bool isCompleted, bool isUpdated,
+            string[] expectedFeeds, string[] notExpectedFeeds)
+        {
+            string targetRepoUri = GetAzDoApiRepoUrl(targetRepoName);
+            TestContext.WriteLine($"Checking Opened PR in {targetBranch} {targetRepoUri} ...");
+            int pullRequestId = await GetAzDoPullRequestIdAsync(targetRepoName, targetBranch);
+            Microsoft.DotNet.DarcLib.PullRequest pullRequest = await GetAzDoPullRequestAsync(pullRequestId, targetRepoName, targetBranch, isUpdated, expectedPRTitle);
+
+            string trimmedTitle = Regex.Replace(pullRequest.Title, @"\s+", " ");
+            StringAssert.AreEqualIgnoringCase(expectedPRTitle, trimmedTitle);
+
+            Microsoft.DotNet.DarcLib.PrStatus expectedPRState = isCompleted ? Microsoft.DotNet.DarcLib.PrStatus.Closed : Microsoft.DotNet.DarcLib.PrStatus.Open;
+            var prStatus = await AzDoClient.GetPullRequestStatusAsync(GetAzDoApiRepoUrl(targetRepoName) + $"/pullRequests/{pullRequestId}");
+            Assert.AreEqual(expectedPRState, prStatus);
+
+            using (ChangeDirectory(repoDirectory))
```

Comment:
Ugh... This "ChangeDirectory" concept sort of means this _entire assembly_ needs `[NonParallelizable]`.  What code uses the Environment.CurrentDirectory... because they really, really shouldn't do that.  It's a fragile piece of code by itself, and it's very hard to test.

Reply:
These tests are 100% not Parallelizable, they call darc and git all over, all of it stateful stuff using the current directory.

Reply:
We could always set the WorkingDirectory when we launch those processes into isolated places. We _don't_ do that, and it's almost certainly an enormous change to do so.

But we _need_ to find some way to parallelize these in the future, they take way too long to execute.

Reply:
Added it to the affected test classes so this is covered for the time being.